libpod/oci_conmon_common.go

```diff
@@ -1038,6 +1038,34 @@ func (r *ConmonOCIRuntime) getLogTag(ctr *Container) (string, error) {
 	return b.String(), nil
 }
 
+func getPreserveFdExtraFiles(preserveFD []uint, preserveFDs uint) (uint, []*os.File, []*os.File) {
+	var filesToClose []*os.File
+	var extraFiles []*os.File
+
+	preserveFDsMap := make(map[uint]struct{})
+	for _, i := range preserveFD {
+		if i > preserveFDs {
+			preserveFDs = i
+		}
+		preserveFDsMap[i] = struct{}{}
+	}
+
+	if preserveFDs > 0 {
+		for fd := 3; fd < int(3+preserveFDs); fd++ {
```

Comment:
Assume `--preserve-fd=3` has been given, will this iterate over fd=3, fd=4, fd=5 ?  The two last iterations would not be needed.

 (I haven't tried it out by building and running the code as right now I'm only able to read the PR)

Reply:
`--preserve-fds` and `--preserve-fd` cannot be used together.

preserve-fds are the FDs to preserve in addition to the std streams, so fd=3 means the fd until 5 included.  If you pass fd=0, only the first 3 are preserved

Reply:
changed to return an error in this case